sbt/sbt-openapi-codegen/src/main/scala/OpenapiCodegenTask.scala

```diff
@@ -0,0 +1,39 @@
+import sbt._, Keys._
+import codegen._
+
+case class OpenapiCodegenTask(
+  inputYaml: File,
+  dir: File,
+  cacheDir: File
+) {
+  import FileInfo.hash
+  import Tracked.inputChanged
+
+  val tempFile = cacheDir / "sbt-openapi-codegen" / s"TapirGeneratedEndpoints.scala"
+  val outFile = dir / "sbt-openapi-codegen" / s"TapirGeneratedEndpoints.scala"
```

Comment:
right, so here shouldn't we use sth like `managedSources` as the base directory?

Reply:
`cacheDir` and `dir` are params here.

```
(((
      swaggerFile,
      sourceManaged,
      streams
    ) flatMap { (
      swaggerFile: File,
      srcDir: File,
      taskStreams: TaskStreams,
    ) =>
      OpenapiCodegenTask(swaggerFile, srcDir, taskStreams.cacheDirectory).file
    }) map (Seq(_))).value
```
Maps the `swaggerFile` key to the first param, `sourceManaged` as `dir` and `streams.cacheDirectory` as `cacheDir`. I think this is ok.

I get the sbt specific parts more or less from [sbt-buildinfo](https://github.com/sbt/sbt-buildinfo/tree/master/src/main/scala/sbtbuildinfo).

Reply:
ok, makes sense, I missed where these parameters come from, thanks

Reply:
one more question - this makes `sbt-openapi-codegen` the package name then, since `dir` is the managed-sources directory?

Reply:
nope, under `target` you will have a `src_managed` dir, the dir and file will generated under that, and after the file generation/write, the compiler takes it and compiles it, and creates a package corresponding to the files "package header" under the `classes`.

Reply:
ok, right, so the source file doesn't follow the package directory naming convention, but it doesn't have to

Reply:
Yapp... As I understand the BuildInfo plugins code, it is mostly bcs of easier file modification checks. (If you write/update the file, it will be recompiled => it will force to recompile all the files that uses it. The whole `inputChanged` hack is to check if we need to rewrite (=> recompile) the generated file, or not.)